### ISSUE TITLE
AlgoliaSearcher: re-introduce a "Submit Search" button

### ISSUE THREAD

**Issue by afercia** [Yoast/yoast-components#285]
Splitting this out from #243

For accessibility and consistency, consider to re-introduce a "Submit Search" button in the Algolia Search. See https://github.com/Yoast/yoast-components/pull/243#issuecomment-328080925

This was discussed previously, see:
> Another important point is the lack of a "submit search" button. We've had an interesting discussion about the Knowledge Base for My Yoast and decided to re-introduce the "submit search" button, see Yoast/my-yoast#689 (comment) related to #198 (Reactify Help Center - KB tab)

**Comment by afercia** [comment]
Worth noting the mockups on https://github.com/Yoast/yoast-components/issues/198 do have a search submit button.

**Comment by IreneStr** [comment]
Fixed in #316
